editor/code_editor.cpp

```diff
@@ -536,6 +528,10 @@ void FindReplaceBar::_show_search(bool p_focus_replace, bool p_show_only) {
 			search_text->set_caret_column(search_text->get_text().length());
 		}
 
+		_update_flags(false);
```

Comment:
Can we just add a `_search_text_changed(get_search_text())` call after `search_text->set_text(text_editor->get_selected_text(0))` ? Force doing a search when the search bar is shown.

Reply:
That would clean the code up but it triggers a search_current unnecessarily, since the the first search match is already selected in this case. I figured that's why the stuff that was already here was inlined in the first place, so I stuck with that.

If you guys think that slight inefficiency is worth the cleaner code I can change it.

Reply:
@jsjtxietian what do you think about my post above?

Reply:
Ah sorry for the late reply. IMHO both are fine, it's a choice between a little readability and performance. I slightly perfer my suggestion because it introduces less change which could potentially introduce less bug.

Reply:
I made that change, with the exception that I left the `_search_text_changed` where the existing result-updating code was in the not-empty check, to avoid doing it unnecessarily and minimize the change.